packages/vanilla-components/src/components/referral-list/referral-list.tsx

```diff
@@ -3,46 +3,221 @@ import { css } from "emotion";
 import { uuid } from "../../utilities";
 import { API } from "../../services/WidgetHost";
 
+/**
+ * @uiName Referral List
+ */
 @Component({
   tag: "sqh-referral-list",
   styleUrl: "referral-list.scss",
 })
 export class ReferralList {
-  // general in dropdown
+  /**
+   * Hide the component
+   *
+   * @uiName Is Hidden
+   * @uiGroup general
+   */
   @Prop() ishidden: boolean;
+  /**
+   * Text inside the paginate more button
+   *
+   * @uiName Paginate More Text
+   * @uiGroup general
+   */
   @Prop() paginatemore: string;
+  /**
+   * Text inside the paginate less button
+   *
+   * @uiName Paginate Less Text
+   * @uiGroup general
+   */
   @Prop() paginateless: string;
+  /**
+   * Text shown when referral list is empty
+   *
+   * @uiName Empty Referrals Text
+   * @uiGroup general
+   */
   @Prop() noreferralsyet: string;
+  /**
+   * Text color of the referred users name
+   *
+   * @uiName Referral Name Color
+   * @uiGroup general
+   */
   @Prop() referralnamecolor: string;
+  /**
+   * Text color of the referral details
+   *
+   * @uiName Referral Text Color
+   * @uiGroup general
+   */
   @Prop() referraltextcolor: string;
+  /**
+   * Text shown when the referred users name is unknown
+   *
+   * @uiName Unknown User Text
+   * @uiGroup general
+   */
   @Prop() unknownuser: string;
+  /**
+   * Show and hide expired referrals
+   *
+   * @uiName Show Expiry
+   * @uiGroup general
+   */
   @Prop() showexpiry: boolean;
+  /**
+   * Show the invoice details for a referral
```

Comment:
What exactly are the `details` is it the customer note on the reward given? should be a bit more detailed